blob/singularity_store.go

```diff
@@ -159,5 +160,22 @@ func (s *SingularityStore) Describe(ctx context.Context, id ID) (*Descriptor, er
 	if err != nil {
 		return nil, err
 	}
-	return s.local.Describe(ctx, decoded)
+	descriptor, err := s.local.Describe(ctx, decoded)
+	if err != nil {
+		return nil, err
+	}
+	deals, err := s.singularityClient.GetFileDeals(ctx, itemID)
+	replicas := make([]Replica, 0, len(deals))
+	for _, deal := range deals {
+		replicas = append(replicas, Replica{
+			// TODO: figure out how to get LastVerified
+			Provider:   deal.Provider,
+			Status:     string(deal.State),
+			Expiration: epochutil.EpochToTime(deal.EndEpoch),
```

Comment:
This assumes mainnet right? We probably want to parameterise this for devnet testing.